help_to_heat/portal/models.py

```diff
@@ -165,3 +165,53 @@ class ScottishEpcRating(utils.TimeStampedModel):
+    potential_energy_efficiency_rating = models.CharField(max_length=4, null=True)
+    built_form = models.CharField(max_length=32, null=True)
+    inspection_date = models.CharField(max_length=16, null=True)
+    local_authority = models.CharField(max_length=64, null=True)
+    constituency = models.CharField(max_length=64, null=True)
+    energy_consumption = models.CharField(max_length=8, null=True)
+    potential_energy_consumption = models.CharField(max_length=8, null=True)
+    co2_emissions = models.CharField(max_length=8, null=True)
+    co2_emissions_per_floor_area = models.CharField(max_length=8, null=True)
+    co2_emissions_potential = models.CharField(max_length=8, null=True)
+    floor_area = models.CharField(max_length=8, null=True)
+    floor_level = models.CharField(max_length=16, null=True)
+    floor_height = models.CharField(max_length=8, null=True)
+    energy_tariff = models.CharField(max_length=32, null=True)
+    mains_gas = models.CharField(max_length=4, null=True)
+    multiple_glazed_proportion = models.CharField(max_length=4, null=True)
+    extension_count = models.CharField(max_length=4, null=True)
+    habitable_room_count = models.CharField(max_length=4, null=True)
+    heated_room_count = models.CharField(max_length=4, null=True)
+    floor_description = models.CharField(max_length=64, null=True)
+    floor_energy_efficiency = models.CharField(max_length=16, null=True)
+    windows_description = models.CharField(max_length=64, null=True)
+    wall_description = models.CharField(max_length=64, null=True)
+    wall_energy_efficiency = models.CharField(max_length=16, null=True)
+    wall_environmental_efficiency = models.CharField(max_length=16, null=True)
+    main_heating_description = models.CharField(max_length=64, null=True)
+    main_heating_energy_efficiency = models.CharField(max_length=16, null=True)
+    main_heating_environmental_efficiency = models.CharField(max_length=16, null=True)
+    main_heating_fuel_type = models.CharField(max_length=64, null=True)
+    second_heating_description = models.CharField(max_length=64, null=True)
+    second_heating_energy_efficiency = models.CharField(max_length=16, null=True)
+    roof_description = models.CharField(max_length=64, null=True)
+    roof_energy_efficiency = models.CharField(max_length=16, null=True)
+    roof_environmental_efficiency = models.CharField(max_length=16, null=True)
+    lighting_description = models.CharField(max_length=64, null=True)
+    lighting_energy_efficiency = models.CharField(max_length=16, null=True)
+    lighting_environmental_efficiency = models.CharField(max_length=16, null=True)
+    mechanical_ventilation = models.CharField(max_length=64, null=True)
+    construction_age_band = models.CharField(max_length=16, null=True)
+    tenure = models.CharField(max_length=64, null=True)
+    improvements = models.TextField(max_length=1024, null=True)
+    alternative_improvements = models.TextField(max_length=1024, null=True)
```

Comment:
Is this long enough?

Reply:
@jdgage 
I'm not sure what the maximum length of these fields can be, since I think they do something similar to us in that this is a collection of all the improvement strings.
I don't think we _need_ to assign one since we're using Postgresql: 
https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.CharField.max_length
I'm not sure how VarChars are dealt with though, would this take up more memory than limited length fields?

Reply:
Would they: yes. Does it matter: probably not a great deal. Generally it's a good idea to put an upper limit on column size, but only if you can be confident about it (and I'm not sure we can in this case?). It matters less if you're not querying against that column, which is the case here.